dist/update-release-notes.go

```diff
@@ -0,0 +1,449 @@
+	authTokenPtr := flag.String("AuthToken", "", "Visit: (https://github.com/settings/tokens) generate a token with 'public_repo' or 'repo' permissions. Store access token in a file (token.txt). Example usage: -AuthToken=$(cat token.txt)")
+	repoPtr := flag.String("Repo", "quick-lint/quick-lint-js", "GitHub repo where release notes to be released.")
+	tagsRepoPtr := flag.String("TagsRepo", "quick-lint/quick-lint-js", "GitHub repo to get release tags from.")
+	flag.Parse()
+	if *authTokenPtr == "" {
+		fmt.Println(redColor + "WARNING: No GitHub access token given for flag -AuthToken. Refer to --help" + resetColor)
+	}
+	return help, authTokenPtr, repoPtr, tagsRepoPtr
+}
+
+func displayHelp(help *bool) {
+	if *help {
+		fmt.Fprintf(os.Stderr, "Usage: %s [OPTIONS]\n", os.Args[0])
+		flag.PrintDefaults()
+		os.Exit(0)
+	}
+}
+
+func getChangeLogPath() string {
+	_, filename, _, _ := runtime.Caller(0)
+	pathToChangeLog := filepath.Join(filepath.Dir(filename), "../docs/CHANGELOG.md")
+	return pathToChangeLog
+}
+
+func handleChannels(channel chan bool, waitGroup *sync.WaitGroup) {
+	timeout := time.After(10 * time.Second)
+	for {
+		select {
+		case <-channel:
+			if len(channel) == 0 {
+				waitGroup.Wait()
+				close(channel)
+				return
+			}
+		case <-timeout:
+			fmt.Println("Timeout: Some goroutines did not complete")
+			close(channel)
+			return
+		}
+	}
+}
+
+func createMissingReleases(releaseMetaData releaseMetaData, authToken string, repoPath string) {
+	createReleaseWaitGroup := &sync.WaitGroup{}
+	createReleaseChannel := make(chan bool, len(releaseMetaData.ReleaseVersionTagMap))
+
+	spawnedThread := false
+	makeLatestRelease := "false"
+	for releaseVersion, tagVersion := range releaseMetaData.ReleaseVersionTagMap {
+		if releaseMetaData.ReleaseVersionTagMap[releaseVersion] == releaseMetaData.TagReleaseVersionMap[tagVersion] {
+			if releaseVersion == releaseMetaData.LatestReleaseVersion {
+				makeLatestRelease = "true"
+			} else {
+				makeLatestRelease = "false"
+			}
+			repoOwner, repoName := splitAndEncodeURLPath(repoPath)
+			requestURL := fmt.Sprintf("https://api.github.com/repos/%v/%v/releases", repoOwner, repoName)
+			postRequest := releaseRequest{
+				authToken:         authToken,
+				repoPath:          repoPath,
+				requestType:       "POST",
+				tagForRelease:     tagVersion,
+				versionTitle:      releaseMetaData.ReleaseVersionTitleMap[releaseVersion],
+				releaseNote:       releaseMetaData.TagVersionReleaseBodyMap[tagVersion],
+				makeLatestRelease: makeLatestRelease,
+			}
+			createReleaseWaitGroup.Add(1)
+			go updateOrCreateGitHubRelease(postRequest, requestURL, createReleaseChannel, createReleaseWaitGroup)
+			spawnedThread = true
+		}
+	}
+	if spawnedThread {
+		handleChannels(createReleaseChannel, createReleaseWaitGroup)
+	} else {
+		close(createReleaseChannel)
+	}
```

Comment:
Why do we have a channel at all? Can't we just call `waitGroup.Wait()` here instead, removing the need for the channels entirely?
```suggestion
	createReleaseWaitGroup.Wait()
```

Reply:
It was unnecessary. :sweat_smile: